### PR TITLE
Kontena Cloud image registry

### PR DESCRIPTION
**Pull request by jakolehm**
Docs for https://github.com/kontena/kontena-plugin-cloud/pull/39

### CODE REVIEW

File: advanced/internal-image-registry.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+---
+title: Internal Image Registry
+---
+
+# Kontena Internal Image Registry
+
+Kontena has built-in support for creating an Internal Image Registry. The Internal Image Registry may be used to store Docker images used by your own application. Kontena's Internal Image Registry runs on the same infrastructure and private network as your Nodes. Therefore, access to Kontena's Image Registry is not open publicly and may be accessed only by using [VPN access](../tools/vpn-access.md). This isolation ensures that you will have total control over the access control, security and distribution of your images. Kontena Image Registry is based on [Docker Image Registry](https://docs.docker.com/registry/).
+
+Kontena may be used with any Docker Image Registry. Users looking for a zero-maintenance, ready-to-go solution are encouraged to consider [Kontena Cloud](/tools/image-registry.md), [Docker Hub](https://hub.docker.com/account/signup/) or [Quay](https://quay.io/). Both of these services provide a hosted registry, as well as some advanced features.
```
Comment (jnummelin) [inline advanced/internal-image-registry.md]: > Both of these services

--> All of these ...

File: advanced/internal-image-registry.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+You should use Kontena's built-in Image Registry if you want to:
+
+* Have total control over where your images are being stored
+* Fully own your images' distribution pipeline
+* Ensure access control and security for your own Docker images
+
```
Comment (jnummelin) [inline advanced/internal-image-registry.md]: > Fully own your images' distribution pipeline
and
> Ensure access control and security for your own Docker images

Can't one accomplish both using KCloud Registry too? I kinda get the point, but these statements might make someone wonder if KCloud registry does not provide access control and security. :)

File: advanced/internal-image-registry.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+## Using Image Registry
```
Comment (jnummelin) [inline advanced/internal-image-registry.md]: Didn't really review from this point forward, I suppose it's just copy-pasta from the old. :)

File: tools/image-registry.md [hunk 4]
```diff
@@ -2,152 +2,155 @@
...
-  * [Local Storage Backend](image-registry.md#local-storage-backend)
-  * [Amazon S3 Storage Backend](image-registry.md#amazon-s3-storage-backend)
-  * [Azure Storage Backend](image-registry.md#azure-storage-backend)
-* [Accessing Image Registry](image-registry.md#accessing-image-registry)
-* [TLS/SSL](image-registry.md#tlsssl)
-* [Authentication](image-registry.md#authentication)
+### Using with Docker CLI
 
-### Create Image Registry Service
+#### Login with Docker CLI
```
Comment (jnummelin) [inline tools/image-registry.md:34]: Placeholder? Or is this actually same as the "Manual way" bit later on?

File: tools/image-registry.md [hunk 5]
```diff
@@ -2,152 +2,155 @@
...
 
-#### Local Storage Backend
+#### Automated way
 
 ```
-$ kontena registry create
+$ kontena cloud image-repository docker-login
```
Comment (jnummelin) [inline tools/image-registry.md]: Might be good to briefly mention what this does.

File: tools/image-registry.md [hunk 6]
```diff
@@ -2,152 +2,155 @@
...
-```
-$ kontena vault write REGISTRY_STORAGE_S3_ACCESSKEY <access_key>
-$ kontena vault write REGISTRY_STORAGE_S3_SECRETKEY <secret_key>
-```
+  ```
+  $ kontena cloud token create <NAME>
+  ```
 
-Create registry service:
+* Login with Docker CLI using your Kontena Cloud username and permanent token from the previous step:
```
Comment (jnummelin) [inline tools/image-registry.md]: Maybe:
...permanent token from the previous step as the password

File: tools/image-registry.md [hunk 7]
```diff
@@ -2,152 +2,155 @@
...
+$ kontena cloud image-repository show <REPOSITORY>
 ```
 
-Write key and certificate to Kontena Vault:
+### Remove a repository
 
 ```
-$ kontena vault write REGISTRY_HTTP_TLS_KEY "$(cat registry_key.pem)"
-$ kontena vault write REGISTRY_HTTP_TLS_CERTIFICATE "$(cat registry_ca.pem)"
+$ kontena cloud image-repository remove <REPOSITORY>
```
Comment (jnummelin) [inline tools/image-registry.md]: Should we mention something that the layers might or might not get deleted? I'm just wondering how many questions we'll get like:
"I removed my repo, but when pushed another not all the layers were pushed."

**Review by jnummelin — CHANGES_REQUESTED** [review]
I think there's couple of things worth "fixing".

File: tools/image-registry.md [hunk 8]
```diff
@@ -2,152 +2,155 @@
...
+### List image repositories
 
 ```
-$ docker build -t registry.<grid_name>.kontena.local/myimage:mytag .
-$ docker push registry.<grid_name>.kontena.local/myimage:mytag
+$ kontena cloud image-repository list --org <ORG>
 ```
 
-Deploying an image from the registry:
+### Create a repository
```
Comment (jnummelin) [inline tools/image-registry.md:84]: We don't have yet `create` command in the PR

Comment (jakolehm) [reply tools/image-registry.md:84]: Should we have?

File: tools/image-registry.md (hunk 7 above)
Comment (jakolehm) [reply tools/image-registry.md]: Idk, might be confusing also to read it here?

File: tools/image-registry.md (hunk 4 above)
Comment (jakolehm) [reply tools/image-registry.md:34]: It should be a header for manual/automated way.. but those sub-headers are confusingly same size. Fixed.

File: tools/image-registry.md (hunk 8 above)
Comment (jnummelin) [reply tools/image-registry.md:84]: Not necessary I think. The first push will create the repo and fire all needed events to track it. At least I don't see missing it as a blocker.

File: tools/image-registry.md (hunk 7 above)
Comment (jnummelin) [reply tools/image-registry.md]: Yeah, might be too confusing

File: tools/image-registry.md (hunk 8 above)
Comment (jakolehm) [reply tools/image-registry.md:84]: IMHO we should add it just to make it logical for users. We can document that a repository can be created with the cli or by just pushing a tag to new repo.

Comment (matti) [reply tools/image-registry.md:84]: create command is not needed imo - just push & forget

**Review by jnummelin — APPROVED** [review]

**Pull request merged**